src/ApplicationInsights.Kubernetes/K8sEnvironment.cs

```diff
@@ -5,34 +5,39 @@
 namespace Microsoft.ApplicationInsights.Kubernetes
 {
     /// <summary>
-    /// Flatten objects for application insights or other external caller to fetch K8s properties.
+    /// An immutable flatten object for application insights or other external caller to fetch K8s properties.
     /// </summary>
-    internal class K8sEnvironment : IK8sEnvironment
+    internal record K8sEnvironment : IK8sEnvironment
     {
-        private readonly V1Pod _pod;
-        private readonly V1ContainerStatus? _containerStatus;
-        private readonly V1ReplicaSet? _replicaSet;
-        private readonly V1Deployment? _deployment;
-        private readonly V1Node? _node;
-
         public K8sEnvironment(
             V1ContainerStatus? containerStatus,
             V1Pod pod,
             V1ReplicaSet? replicaSet,
             V1Deployment? deployment,
-            V1Node? node
-            )
+            V1Node? node)
         {
-            _pod = pod ?? throw new System.ArgumentNullException(nameof(pod));
+            // Pod is required
+            _ = pod ?? throw new System.ArgumentNullException(nameof(pod));
```

Comment:
`<rant>`
We used to write in a simpler style:

```csharp
if (pod is null) {
    throw new ArgumentNullException(nameof(pod));
}
```
and I miss these days 😑 
`</rant>`

Reply:
> `<rant>` We used to write in a simpler style:
> 
> ```cs
> if (pod is null) {
>     throw new ArgumentNullException(nameof(pod));
> }
> ```
> 
> and I miss these days 😑 `</rant>`

I thought it was generated by VSCode, it was me having pod assigned before and removed it.

Regen by code and I got exactly what you have there. I'll take this one.